synapse/storage/database.py

```diff
@@ -2527,16 +2527,16 @@ def simple_search_list_txn(
             retcols: the names of the columns to return
 
         Returns:
-            None if no term is given, otherwise a list of dictionaries.
+            A list of tuples.
         """
         if term:
             sql = "SELECT %s FROM %s WHERE %s LIKE ?" % (", ".join(retcols), table, col)
             termvalues = ["%%" + term + "%%"]
```

Comment:
Aside: I know this is only used in the admin API, so perf is not a top concern... but I can't see how the server can satisfy this query without doing a sequential scan... unless the WHERE conditions are very specific.

Reply:
I kind of wonder if we should just remove this method and inline the SQL into the `search_users` method?

Reply:
I went with this.